lib/create-subdomain/service.ts

```diff
@@ -38,168 +38,215 @@ interface CreateSubdomainArgs {
+ * Receives an encodeFunctionData function as an argument.
+ */
+export async function executeUniversalResolverCall({
   client,
+  universalResolverContractAddress,
+  dnsName,
+  calldata, //-> hex (da enconded date antes)
   chain,
-}: CreateSubdomainArgs) => {
-  const calls: Hash[] = [];
+  signerAddress,
+  encodeFunctionData,
+}: ExecuteUniversalResolverCallArgs): Promise<{ ok: boolean } | void> {
+  try {
+    await client.readContract({
+      address: universalResolverContractAddress,
+      abi: universalResolver,
+      functionName: "resolve",
+      args: [
+        dnsName,
+        encodeFunctionData({
+          functionName: "getOperationHandler",
+          abi: offchainResolver,
+          args: [encodeFunctionData(calldata)],
+        }),
+      ],
+    });
+  } catch (error) {
+    const data = getRevertErrorData(error);
+    if (!data || !data.args || data.args.length === 0) {
+      console.log({ error });
+      return;
+    }
 
-  if (website) {
-    const websiteCallData = encodeFunctionData({
-      functionName: "setText",
+    const [params] = data.args;
+    const errorResult = decodeErrorResult({
       abi: L1ResolverABI,
-      args: [namehash(name), "url", website],
+      data: params as Hex,
     });
 
-    calls.push(websiteCallData);
-  }
+    switch (errorResult?.errorName) {
+      case "OperationHandledOffchain": {
+        const [domain, url, message] = errorResult.args as [
+          DomainData,
+          string,
+          MessageData,
+        ];
+        await handleDBStorage({
+          domain,
+          url,
+          message,
+          authenticatedAddress: signerAddress,
+          chain,
+        });
 
-  if (description) {
-    const descriptionCallData = encodeFunctionData({
-      functionName: "setText",
-      abi: L1ResolverABI,
-      args: [namehash(name), "description", description],
-    });
+        return { ok: true };
+      }
+      case "OperationHandledOnchain": {
+        const [chainId, contractAddress] = errorResult.args as [
+          bigint,
+          `0x${string}`,
+        ];
 
-    calls.push(descriptionCallData);
-  }
+        // Create a new client for L2 using window.ethereum
+        const l2Client = createPublicClient({
+          chain: getChain(Number(chainId)),
+          transport: custom(window.ethereum),
+        }).extend(walletActions);
 
-  // set address
-  if (address) {
-    const coder = getCoderByCoinName("eth");
-    const addressEncoded = fromBytes(coder.decode(address), "hex");
+        const chainIdNumber = Number(chainId);
 
-    const addressCallData = encodeFunctionData({
-      functionName: "setAddr",
-      abi: L1ResolverABI,
-      args: [namehash(name), 60, addressEncoded],
-    });
+        // Add chain to wallet before switching
+        try {
+          await l2Client.addChain({
+            chain: getChain(chainIdNumber),
+          });
+        } catch (err) {
+          console.error("Failed to add chain:", err);
+        }
```

Comment:
this could be done as soon as the modal of subdomain registration starts with a message like `A wallet connected to Arbitrum is required to register this domain`